tests/test_csv_dataset.py

```diff
@@ -161,6 +161,47 @@ def prepare_csv_file(data, filepath):
             self.assertEqual(type(dataset[-1]["ehr_1"]), int)
             np.testing.assert_allclose(dataset[-1]["ehr_9"], 3.3537, rtol=1e-2)
 
+            # test pre-loaded DataFrame
+            df = pd.read_csv(filepath1)
+            dataset = CSVDataset(src=df)
+            self.assertDictEqual(
+                {k: round(v, 4) if not isinstance(v, str) else v for k, v in dataset[2].items()},
+                {
+                    "subject_id": "s000002",
+                    "label": 4,
+                    "image": "./imgs/s000002.png",
+                    "ehr_0": 3.7725,
+                    "ehr_1": 4.2118,
+                    "ehr_2": 4.6353,
+                },
+            )
+
+            # test pre-loaded multiple DataFrames, join tables with kwargs
+            dfs = [pd.read_csv(i) for i in filepaths]
+            dataset = CSVDataset(src=dfs, on="subject_id")
+            self.assertDictEqual(
+                {k: round(v, 4) if not isinstance(v, (str, np.bool_)) else v for k, v in dataset[3].items()},
+                {
+                    "subject_id": "s000003",
+                    "label": 1,
+                    "image": "./imgs/s000003.png",
+                    "ehr_0": 3.3333,
+                    "ehr_1": 3.2353,
+                    "ehr_2": 3.4000,
+                    "ehr_3": 3.1647,
+                    "ehr_4": 3.0863,
+                    "ehr_5": 3.7255,
+                    "ehr_6": 3.6980,
+                    "ehr_7": 3.6980,
```

Comment:
would be great to test with various data types, otherwise I feel these are redundant.

Reply:
Sure, changed to only test `str`, `int`, `float`, `bool` results in the latest commit.

Thanks.